OneSignalSDK/onesignal/src/main/java/com/onesignal/UserStateSynchronizer.java

```diff
@@ -321,6 +331,49 @@ void onSuccess(String response) {
         });
     }
 
+    private void doSMSLogout(String userId) {
+        String urlStr = "players/" + userId + "/sms_logout";
+        JSONObject jsonBody = new JSONObject();
+        try {
+            ImmutableJSONObject dependValues = currentUserState.getDependValues();
+            if (dependValues.has(SMS_AUTH_HASH_KEY))
+                jsonBody.put(SMS_AUTH_HASH_KEY, dependValues.optString(SMS_AUTH_HASH_KEY));
+
+            ImmutableJSONObject syncValues = currentUserState.getSyncValues();
+            if (syncValues.has(EXTERNAL_USER_ID))
+                jsonBody.put(EXTERNAL_USER_ID, syncValues.optString(EXTERNAL_USER_ID));
+
+            if (syncValues.has(EXTERNAL_USER_ID_AUTH_HASH))
+                jsonBody.put(EXTERNAL_USER_ID_AUTH_HASH, syncValues.optString(EXTERNAL_USER_ID_AUTH_HASH));
+
+            jsonBody.put(APP_ID, syncValues.optString(APP_ID));
+        } catch (JSONException e) {
+            e.printStackTrace();
+        }
+
+        OneSignalRestClient.postSync(urlStr, jsonBody, new OneSignalRestClient.ResponseHandler() {
+            @Override
+            void onFailure(int statusCode, String response, Throwable throwable) {
+                OneSignal.Log(OneSignal.LOG_LEVEL.WARN, "Failed last request. statusCode: " + statusCode + "\nresponse: " + response);
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.